Sources/IndexStoreDB/IndexStoreDB.swift

```diff
@@ -80,6 +80,33 @@ public final class IndexStoreDB {
       return body(SymbolOccurrence(occur))
     }
   }
+
+  public func foreachCanonicalSymbolOccurrenceByName(symbolName: String, body: @escaping (SymbolOccurrence) -> Bool) {
+    indexstoredb_for_each_canonical_symbol_occurence_by_name(impl, symbolName) { (occur) -> Bool in
+      body(SymbolOccurrence(occur))
+    }
+  }
+
+  public func findSymbols(matching query: String) -> [SymbolOccurrence] {
```

Comment:
There's already an indexstoredb function to do a search via a pattern, it is `foreachCanonicalSymbolOccurrenceContainingPattern`. You should expose and use that on the sourcekit-lsp side.